oqupy/bath_correlations.py

```diff
@@ -442,6 +459,24 @@ def __init__(
         assert isinstance(alt_integrator, bool)
         self._alt_integrator = alt_integrator
 
+        # create the first_cutoff function and input check for num_oscillations
```

Comment:
Do we want to assert that, if num_oscillations is provided, alt_integrator should be true? Or at least provide a warning that it will be discarded when alt_integrator is False?